src/frameworks/next/utils.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Whether the given path has a regex or not.
+ * According to the Next.js documentation:
+ * ```md
+ *  To match a regex path you can wrap the regex in parentheses
+ *  after a parameter, for example /post/:slug(\\d{1,}) will match /post/123
+ *  but not /post/abc.
+ * ```
+ * See: https://nextjs.org/docs/api-reference/next.config.js/redirects#regex-path-matching
+ */
+export function pathHasRegex(path: string): boolean {
+  for (let i = 0; i < path.length; i++) {
```

Comment:
This would probably be more readable, tunable with a regex, mind converting?

Reply:
@TheIronDev in the last commit I pushed a working regex for that.. I think it can get tricky with `.includes` because of paths like this:
```
/post/\\(escapedparentheses\\)/:slug(\\d{1,}) <--- this has a regex and escaped parenthesis
```